project-auth/src/main/scala/io/renku/projectauth/ProjectAuthService.scala

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2023 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.projectauth
+
+import cats.MonadThrow
+import cats.data.NonEmptyList
+import cats.effect._
+import cats.syntax.all._
+import fs2.io.net.Network
+import fs2.{Pipe, Stream}
+import io.circe.Decoder
+import io.renku.graph.model.projects.{Slug, Visibility}
+import io.renku.graph.model.{RenkuUrl, Schemas}
+import io.renku.jsonld.NamedGraph
+import io.renku.jsonld.syntax._
+import io.renku.tinytypes.json.TinyTypeDecoders._
+import io.renku.triplesstore.client.http.{ConnectionConfig, RowDecoder, SparqlClient}
+import io.renku.triplesstore.client.syntax._
+import org.typelevel.log4cats.Logger
+
+import scala.concurrent.duration._
+
+/** Manage authorization data for projects and members. */
+trait ProjectAuthService[F[_]] {
+
+  def update(data: ProjectAuthData): F[Unit]
+
+  def updateAll: Pipe[F, ProjectAuthData, Nothing]
+
+  def remove(slugs: NonEmptyList[Slug]): F[Unit]
+  def remove(slug:  Slug, more: Slug*): F[Unit] = remove(NonEmptyList(slug, more.toList))
+
+  def getAll(chunkSize: Int = 100): Stream[F, ProjectAuthData]
+}
+
+object ProjectAuthService {
+
+  def resource[F[_]: Async: Network: Logger](
+      connectionConfig: ConnectionConfig,
+      timeout:          Duration = 20.minutes
+  )(implicit renkuUrl: RenkuUrl): Resource[F, ProjectAuthService[F]] =
+    SparqlClient(connectionConfig, timeout).map(c => apply[F](c, renkuUrl))
+
+  def apply[F[_]: MonadThrow](client: SparqlClient[F], renkuUrl: RenkuUrl): ProjectAuthService[F] =
+    new Impl[F](client, renkuUrl)
+
+  private final class Impl[F[_]: MonadThrow](sparqlClient: SparqlClient[F], renkuUrl: RenkuUrl)
+      extends ProjectAuthService[F] {
+    private[this] val graph = Schemas.renku / "ProjectAuth"
+    private implicit val rUrl: RenkuUrl = renkuUrl
+
+    override def remove(slugs: NonEmptyList[Slug]): F[Unit] =
+      sparqlClient.update(sparql"""PREFIX schema: <http://schema.org/>
+                                  |PREFIX renku: <https://swissdatasciencecenter.github.io/renku-ontology#>
+                                  |
+                                  |DELETE { Graph ${graph.asSparql} {?s ?p ?o} }
```

Comment:
It's a super tiny thing but I believe you don't need the `.asSparql` on a `Property`. Or from a different side, you we need it, I'd say we should tweak the `sparql` String Interpolator to be able to work with it.